app/src/main/shared/components/Login/LoggedInUserBox.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import { logOut } from "../../actions/_AuthActions";
+import { InternalLink } from "../InternalLink";
+import { connect } from "react-redux";
+
+import "./Logout.scss";
+
+export class LoggedInUserBoxComponent extends React.Component<any, undefined> {
+    constructor() {
+        super();
+        this.logout = this.logout.bind(this);
+    }
+
+    logout(e: React.MouseEvent<HTMLAnchorElement>) {
+        this.props.dispatch(logOut());
+    }
+
+    render() :JSX.Element {
+        if (this.props.auth.loggedIn) {
+            return <div className="logout">
+                <div>
+                    Logged in as { this.props.auth.username } |&nbsp;
+                    <InternalLink href="/" onClick={ this.logout }>
+                        Log out
+                    </InternalLink>
+                </div>
+            </div>;
+        } else {
+            return null;
+        }
+    }
+}
+
+const mapStateToProps = (state: any) => ({
```

Comment:
we can add `GlobalState` as the type here, and then, map

```
const mapStateToProps = (state: GlobalState): Partial<LoggedInUserBoxProps> => ({
    loggedIn: state.auth.loggedIn,
    username: state.auth.username
});
```
Then we could also do:

```
const mapDispatchToProps = (dispatch: Dispatch<Unauthenticated>): Partial<LoggedInUserBoxProps> => ({
    logOut: () => dispatch(logOut())
});
```

as per this example:
https://www.sohamkamani.com/blog/2017/03/31/react-redux-connect-explained/